### PR TITLE
First Visualization

### PR DESCRIPTION
**Pull request by marcusnovotny**
Hey there! This is my first visualization. Still a lot to do, but I need some feedback. 

About the files:
- site.html: App centering around a canvas element. The user can input a GoT Character name and a date range (only works in Chrome as of now) and gets back a graph of our data + trend function. I haven't implemented the regression yet and the db doesn't have the needed data atm, so it's currently showing fake data and a log x function as a trend curve.
- main.css: Very ugly, fixed layout for development purposes
- basic.js: Initializes the date elements. Sets up Event Handlers. Checks user input. Collects Data from getData.js, supplies it to plot.js
- getData.js: Currently used to generate and acces mockup data for the graph. Should later handle the db connection
- plot.js: Pretty messy script as of now - Draws the fake data curve + a trend function (currently just a log x) on the canvas. Adds dashed lines to show days.

Missing features:
- There's zero information on x & y axis right now. I thought it might be easier to display that in a tooltip when hovering over the canvas - that's missing at the moment though.
- Regression function calculation for the trend curve
- Autocomplete / Suggestion Box for the Searchbar based upon Characters in the DB
- Highlighting Episode airing dates
- Other fancy stuff

Also wanna add JQuery for some things. Looking forward to your feedback, I'll get some sleep now.

### CODE REVIEW

File: public/site.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            <input type="search" name="charname" maxlength=50 placeholder="GoT Character" required="" id="searchBar" />
+            <input type="submit" value="Search" id="submitButton" />
+        </form>
+        <canvas id="canvas" width=770 height=300>
+            <!-- Width and height have to be specified here in order to prevent blurryness-->
+            <div id="nocanvas">
+                <p>It seems like your browser does not support canvas, a common HTML5 element.
+                    <br> Please update your browser and come back here!
+                    <br>
+                    <img src="http://hellogiggles.com/wp-content/uploads/2015/08/12/kit-harington-sad-jon-snow-game-of-thrones.jpg" width=450 height=225/></p>
```
Comment (julienschmidt) [inline public/site.html]: hot-linking is not so nice. Add to image to public/img instead :wink:

**Comment by julienschmidt** [comment]
Looking good so far!

Probably it is a good idea to use a library like [D3.js](https://d3js.org/). Otherwise this part will become cumbersome (no need to reinvent the wheel).

Some further hints:
- the y-axis should at least indicate the dimension and a rough scale. Right now I can't tell if this graph is based on 5 or 5 million tweets.
- the graph should show a colored area for both positive tweets (above the x-axis, light green color) and negative tweets (below x-axis, light red color)
- when hovered, the graph should indicate the current absolute value at the given point on the x-axis. Existing libraries should help here.
- the most popular characters should be suggested (the backend will provide an API call for that). Some people might not know any characters (like me :stuck_out_tongue_closed_eyes:)
- an option (when we're done with the basic stuff) to compare characters would be really, really nice

Maybe delete the app.html and then press the big green merge button :+1:

**Comment by gyachdav** [comment]
Is there a demo we can see? 

Sent from my iPhone

**Comment by julienschmidt** [comment]
For now, checkout the branch (`frontend`) and open public/site.html in Chrome (Firefox doesn't like the date input, Marcus is on it already)

**Pull request merged**